### DISCUSSION TITLE
Print all the files selected in the explorer pane

### DISCUSSION THREAD

**Discussion by manuel-rw** [PDConSec/vsc-print#226] [Ideas]
Hello,
thank you for this project.
I am having an issue, which keeps me from using it sadly.

**Reproduce**:
- Install the extension
- Create a project with multiple code files (eg. *.cs)
- Click the first file with a left click
- Press and hold SHIFT
- Click the last file you want to print (eg. the 4.)
- Release SHIFT
- Right click (notice how multiple files are selected)
- Click print

**Expected behavior**:
All selected files are printed, regardless which file has been last clicked on

**Actual behavior**:
Only the last file, that has been clicked, is printed.
In this case, it would be the 4.

I saw #112 and #162, but it seems like they both are about selections in the code itself, and not in the file explorer of VSC.
Let me know if you have questiond or if I should provide more information.

**Comment by PeterWone** [comment]
The behaviour you seem to want is for the file print context menu for the explorer to print the files indicated by multiple selection in the file explorer. 

This is not a currently supported feature because you're the first to ask for it. I will look into it.

In the interim you can gather the required files into a folder and print that.

**Comment by manuel-rw** [comment]
Hi @PeterWone ,
thanks for the reply.
Since I need to keep the original paths, that's not going to work.
But I don't mind waiting until you can select multiple items.
Thanks for you work :)

**Comment by PeterWone** [comment]
Despite utterly dismal documentation I have figured out how to get a list of the selected files. I should be able to do this. I will use the same style as folder printing - one long print job with file path headings between files.

**Comment by PeterWone** [comment]
OK, here it is. Same style as folder printing, but from a multi-selection. This will appear in the next release which I am trying to prepare right now.

![image](https://user-images.githubusercontent.com/5498936/221391116-eb9f0f51-edbd-4716-be3a-030d65ce58ac.png)

Observe that three files were selected but the toxic one was ignored (`.design` is a binary file for vector graphics)

![image](https://user-images.githubusercontent.com/5498936/221391258-16857d41-fd11-40c0-bdeb-ea2197fc9bb4.png)
